platform/ui/src/utils/providers/DialogProvider.js

```diff
@@ -0,0 +1,164 @@
+import React, {
+  useState,
+  createContext,
+  useContext,
+  useCallback,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import Draggable from 'react-draggable';
+
+const DialogContext = createContext(null);
+
+export const useDialog = () => useContext(DialogContext);
+
+const DialogProvider = ({ children, service }) => {
+  const [count, setCount] = useState(1);
+  const [dialogs, setDialogs] = useState([]);
+
+  /**
+   * Sets the implementation of a dialog service that can be used by extensions.
+   *
+   * @returns void
+   */
+  useEffect(() => {
+    if (service) {
+      service.setServiceImplementation({ create, dismiss, dismissAll });
+    }
+  }, [create, dismiss, service]);
+
+  /**
+   * Creates a dialog and return its id.
+   *
+   * @returns id
+   */
+  const create = useCallback(
+    ({ id, content, ...props }) => {
+      let dialogId = id;
+      if (!dialogId) {
+        dialogId = Math.random()
+          .toString(36)
+          .substr(2, 5);
```

Comment:
Oh, interesting. Allow them to provide their own ID. I like it.
For random, do we have a GUID generator? Could use incrementing count, but would need to note that custom IDs should not be numbers as string to prevent conflicts.